src/bosh_openstack_cpi/lib/cloud/openstack/helpers.rb

```diff
@@ -123,11 +123,16 @@ def wait_resource(resource, target_state, state_method = :status, allow_notfound
         # If resource reload is nil, perhaps it's because resource went away
         # (ie: a destroy operation). Don't raise an exception if this is
         # expected (allow_notfound).
-        if with_openstack { resource.reload.nil? }
-          break if allow_notfound
+        begin
+          if with_openstack { resource.reload.nil? }
+            break if allow_notfound
+            cloud_error("#{desc}: Resource not found")
+          else
+            state =  with_openstack { resource.send(state_method).downcase.to_sym }
+          end
+        rescue Bosh::Clouds::CloudError => e
```

Comment:
Whoops, not sure why I added this originally. I think we were seeing errors here while we were still getting our fog monkeypatch figured out. Doesn't seem to be needed, fixed here: https://github.com/cloudfoundry-incubator/bosh-openstack-cpi-release/pull/49/commits/c33f8c0827528633d1a89d39d3fa12025ef3024d